src/core/backbone_router/local.cpp

```diff
@@ -238,6 +241,92 @@ void Local::UpdateBackboneRouterPrimary(Leader::State aState, const BackboneRout
     return;
 }
 
+otError Local::GetDomainPrefix(NetworkData::OnMeshPrefixConfig &aConfig)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(mDomainPrefixConfig.mPrefix.mLength > 0, error = OT_ERROR_NOT_FOUND);
+
+    aConfig = mDomainPrefixConfig;
+
+exit:
+    return error;
+}
+
+otError Local::RemoveDomainPrefix(const otIp6Prefix &aPrefix)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(aPrefix.mLength > 0, error = OT_ERROR_INVALID_ARGS);
+
+    VerifyOrExit(mDomainPrefixConfig.mPrefix.mLength == aPrefix.mLength, error = OT_ERROR_NOT_FOUND);
+
+    VerifyOrExit(otIp6PrefixMatch(&mDomainPrefixConfig.mPrefix.mPrefix, &aPrefix.mPrefix) >=
+                     mDomainPrefixConfig.mPrefix.mLength,
+                 error = OT_ERROR_NOT_FOUND);
+
+    mDomainPrefixConfig.mPrefix.mLength = 0;
+
+    if (IsEnabled())
+    {
+        SuccessOrExit(error =
+                          Get<NetworkData::Local>().RemoveOnMeshPrefix(aPrefix.mPrefix.mFields.m8, aPrefix.mLength));
+        Get<NetworkData::Notifier>().HandleServerDataUpdated();
+    }
+
+exit:
+    return error;
+}
+
+otError Local::SetDomainPrefix(const NetworkData::OnMeshPrefixConfig &aConfig)
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(aConfig.mPrefix.mLength > 0, error = OT_ERROR_INVALID_ARGS);
+
+    if (IsEnabled())
+    {
+        Get<NetworkData::Local>().RemoveOnMeshPrefix(mDomainPrefixConfig.mPrefix.mPrefix.mFields.m8,
+                                                     mDomainPrefixConfig.mPrefix.mLength);
+    }
+
+    mDomainPrefixConfig = aConfig;
+
+    if (IsEnabled())
+    {
+        SuccessOrExit(error = Get<NetworkData::Local>().AddOnMeshPrefix(mDomainPrefixConfig));
```

Comment:
Any `aConfig` that fail in this step will leave `mDomainPrefixConfig` in an inconsistent state because `mDomainPrefixConfig.mPrefix.mLength > 0` but it's not in local data, like the MLEID `fdde:ad00:beef:0::/64` above.